bundler/lib/bundler/dep_proxy.rb

```diff
@@ -4,18 +4,27 @@ module Bundler
   class DepProxy
     attr_reader :__platform, :dep
 
+    @proxies = {}
+
+    def self.get_proxy(dep, platform)
+      @proxies[[dep, platform].hash] ||= new(dep, platform)
+    end
+
     def initialize(dep, platform)
       @dep = dep
       @__platform = platform
     end
 
-    def hash
-      @hash ||= [dep, __platform].hash
+    private_class_method :new
+    private :dup
+
+    # reset before calling
+    def self.setup!
+      @proxies = {}
     end
 
-    def ==(other)
-      return false if other.class != self.class
-      dep == other.dep && __platform == other.__platform
+    def hash
+      @hash ||= [dep, __platform].hash
```

Comment:
If I understood your description correctly, molinillo now always does "object based comparison", so the specific hash ruby (or us) assigns to the object should be irrelevant since it's no longer used (and even if molinillo still used it somewhere for comparisons, it will still work since same objects should have the same default `#hash`, correct? In order words, do we need this method?

Reply:
You are right, especially according to this definition: https://stackoverflow.com/questions/1791171/ruby-object-hash. I'll remove it.

Reply:
Removed in 37c6a7e4c273f9c8cfaeaef7b56a011521497f0d

Reply:
Thanks :heart:!